### PR TITLE
Homework5 task3 - Vacations - ready

### PR DESCRIPTION
**Pull request by VytasR**
Homework5 task3 - Vacations - ready for review

### CODE REVIEW

File: Homework5/VacationsApp/CompanyEntities/Company.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            foreach (var group in vacationGroupsbyEmployee)
+            {
+                foreach (var item in group.vacation)
+                {
+                    var firstDay = item.FirstDay;
+                    var lastDay = item.LastDay;
+
+                    foreach (var vacationToCompare in group.vacation)
+                    {
+                        if (item != vacationToCompare && !result.Contains((item, vacationToCompare)) &&
```
Comment (radislavkireev) [inline Homework5/VacationsApp/CompanyEntities/Company.cs:93]: You may have possible duplicates in result. 

Consider case when group.vacation contains 2 records with the same First and Last days.
Lets name them v1 and v2. 

After "foreach" result may contain  (v1,v2) and (v2,v1) tuples. They are equal form users perspective, but result.Contains does hot catch it. 

I have not run the code, pls run the tests to check if proposal is valid.

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: Homework5/VacationsApp/CompanyEntities/Company.cs (hunk 1 above)
Comment (VytasR) [reply Homework5/VacationsApp/CompanyEntities/Company.cs]: Thanks for the remark. I have run the tests. If there are 2 Vacation records in the group with the same First and Last days, method returns one (v1, v2) tuple. If there are 3 analogous records, method returns three tuples: (v1, v2), (v1, v3), (v2, v3).

Comment (VytasR) [reply Homework5/VacationsApp/CompanyEntities/Company.cs]: After today's lecture I wrote the code a bit more elegantly with LINQ. Had to check !result.Contains(v2, v1) before adding to the result. Then no duplicates appear.

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: Homework5/VacationsApp/CompanyEntities/Company.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            var employeeVacationMonths = new HashSet<(string, int)>();
+            foreach (var vacation in _vacations)
+            {
+                employeeVacationMonths.Add((vacation.EmployeeName, vacation.FirstDay.Month));
+                employeeVacationMonths.Add((vacation.EmployeeName, vacation.LastDay.Month));
+            }   
+
+            var monthsNumberOfEmployees = employeeVacationMonths.GroupBy(entry => entry.Item2).Select(group => (group.Key, group.Count()));
+            var months = Enumerable.Range(1, 12); 
+            var result = from month in months
```
[the remark this answers is not in the document]
Comment (VytasR) [reply Homework5/VacationsApp/CompanyEntities/Company.cs:53]: Fixed.

**Review by radislavkireev — APPROVED** [review]

**Pull request merged**